device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -429,4 +425,25 @@ public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeC
         //Codes_SRS_DEVICE_IO_34_020: [This function shall register the callback with the transport.]
         this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext);
     }
+
+    /*
+     * IotHubTransport layer will notify this layer when the connection is established and when it is lost. This layer should start/stop
+     * the send/receive threads accordingly
+     */
+    @Override
+    public void execute(IotHubConnectionStatus status, IotHubConnectionStatusChangeReason statusChangeReason, Throwable throwable, Object callbackContext)
```

Comment:
this method will be called back from the transport layer every time the state changes. This class should start/stop the send/receive thread pool and then update its own state to be consistent with the transport state